src/modules/schnorr/schnorr_impl.h

```diff
@@ -16,89 +16,84 @@
- * batch validation offers potentially much higher benefits than this cost.
- *
- * Rationale for having an implicit y coordinate oddness:
- * If we commit to having the full R point known to verifiers, there are two mechanism.
- * Either include its oddness in the signature, or give it an implicit fixed value.
- * As the R y coordinate can be flipped by a simple negation of the nonce, we choose the
- * latter, as it comes with nearly zero impact on signing or validation performance, and
- * saves a byte in the signature.
- *
- * Signing:
- *   Inputs: 32-byte message m, 32-byte scalar key x (!=0), 32-byte scalar nonce k (!=0)
- *
- *   Compute point R = k * G. Reject nonce if R's y coordinate is odd (or negate nonce).
- *   Compute 32-byte r, the serialization of R's x coordinate.
- *   Compute scalar h = Hash(r || m). Reject nonce if h == 0 or h >= order.
- *   Compute scalar s = k - h * x.
- *   The signature is (r, s).
- *
- *
- * Verification:
- *   Inputs: 32-byte message m, public key point Q, signature: (32-byte r, scalar s)
- *
- *   Signature is invalid if s >= order.
- *   Signature is invalid if r >= p.
- *   Compute scalar h = Hash(r || m). Signature is invalid if h == 0 or h >= order.
- *   Option 1 (faster for single verification):
- *     Compute point R = h * Q + s * G. Signature is invalid if R is infinity or R's y coordinate is odd.
- *     Signature is valid if the serialization of R's x coordinate equals r.
- *   Option 2 (allows batch validation and pubkey recovery):
- *     Decompress x coordinate r into point R, with odd y coordinate. Fail if R is not on the curve.
- *     Signature is valid if R + h * Q + s * G == 0.
- */
-
-static int secp256k1_schnorr_sig_sign(const secp256k1_ecmult_gen_context* ctx, unsigned char *sig64, const secp256k1_scalar *key, const secp256k1_scalar *nonce, const secp256k1_ge *pubnonce, secp256k1_schnorr_msghash hash, const unsigned char *msg32) {
-    secp256k1_gej Rj;
-    secp256k1_ge Ra;
+static void secp256k1_schnorr_ge_get_b32(unsigned char *b32, secp256k1_ge* p) {
+    secp256k1_fe_normalize(&p->x);
+    secp256k1_fe_get_b32(b32, &p->x);
+}
+
+static int secp256k1_schnorr_ge_set_b32(secp256k1_ge* p, const unsigned char *b32) {
+    secp256k1_fe x;
+    if (!secp256k1_fe_set_b32(&x, b32)) {
+        return 0;
+    }
+    return secp256k1_ge_set_xo_var(p, &x, 0);
+}
+
+/** Computes {priv = +/- (scalar)b32; pub = priv * G } with pub.y even. */
+static int secp256k1_schnorr_nonces_set_b32(const secp256k1_ecmult_gen_context* ctx, secp256k1_scalar* priv, secp256k1_ge* pub, const unsigned char *b32, const secp256k1_ge* pub_others) {
+    int overflow = 0;
+    int flip = 0;
+    secp256k1_gej gej;
+
+    secp256k1_scalar_set_b32(priv, b32, &overflow);
+    if (overflow || secp256k1_scalar_is_zero(priv)) {
+        secp256k1_scalar_clear(priv);
+        return 0;
+    }
+    secp256k1_ecmult_gen(ctx, &gej, priv);
+    if (secp256k1_gej_is_infinity(&gej)) {
+        return 0;
```

Comment:
Can this branch be hit? If `priv` is zero the above check gets it; if `priv` is n (or higher) then `overflow` should be set and the above check gets it.